inference-engine/tests/functional/plugin/gpu/shared_tests_instances/single_layer_tests/tensor_iterator.cpp

```diff
@@ -0,0 +1,296 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <string>
+#include <memory>
+#include <utility>
+#include <map>
+#include <vector>
+#include <ngraph/op/util/attr_types.hpp>
+#include <cldnn/cldnn_config.hpp>
+#include <transformations/control_flow/unroll_tensor_iterator.hpp>
+#include "common_test_utils/test_constants.hpp"
+#include "ie_api.h"
+#include "shared_test_classes/base/layer_test_utils.hpp"
+#include "ngraph_functions/builders.hpp"
+#include "ngraph_functions/utils/ngraph_helpers.hpp"
+
+using namespace InferenceEngine;
+using Config = std::pair<std::string, std::map<std::string, std::string>>;
+
+namespace LayerTestsDefinitions {
+
+using TensorIteratorWithConfigParams = typename std::tuple<
+        size_t,                                 // seq_lengths
+        size_t,                                 // batch
+        size_t,                                 // hidden size
+        // todo: fix. input size hardcoded to 10 due to limitation (10 args) of gtests Combine() func.
+        //size_t,                               // input size
+        size_t,                                 // sequence axis
+        float,                                  // clip
+        ngraph::helpers::TensorIteratorBody,    // body type
+        ngraph::op::RecurrentSequenceDirection, // direction
+        InferenceEngine::Precision,             // Network precision
+        Config>;                                // Target device name & Configuration
+
+class TensorIteratorWithConfigTest : public testing::WithParamInterface<TensorIteratorWithConfigParams>,
```

Comment:
If it's a common tests for all plugins, the please move test definition to shared folder. If it's GPU specific tests, then please move it out of shared_tests_instances folder for gpu

Reply:
The testcase is moved to `tests/functional/plugin/gpu/single_layer_tests/tensor_iterator.cpp`